File: program/program/find-sum-of-gp-series/FindSumOfGpSeries.kt
fun main() {
    val a1 = 2 // First term of the GP
    val r = 3  // Common ratio of the GP
    val n = 3  // Number of terms in the series

    val sum = findSumOfGP(a1, r, n)
    println("Sum of the GP series: $sum")
}

fun findSumOfGP(a1: Int, r: Int, n: Int): Int {
    // Formula to find the sum of a GP: S = a * (1 - r^n) / (1 - r)
    val sum = a1 * (1 - Math.pow(r.toDouble(), n.toDouble()).toInt()) / (1 - r)
    return sum
}

File: program/program/find-sum-of-gp-series/find_sum_of_gp_series.dart
import 'dart:math';

void main() {
  
  int a1 = 2; // First term of the GP
  int r = 3; // Common ratio
  int n = 3; // Number of terms

  // Calculating the sum of the GP series
  double sum = calculateGPSum(a1, r, n);

  // Printing the output
  print("Sum of GP series: $sum");
}

double calculateGPSum(int a1, int r, int n) {
  // Using the formula for the sum of GP series
  return a1 * (pow(r, n) - 1) / (r - 1);
}
